### DISCUSSION TITLE
Reset demo from beginning

### DISCUSSION THREAD

**Discussion by Henroks** [saul/demofile#261] [Questions] [answered]
**Is your feature request related to a problem? Please describe.**
No

**Describe the solution you'd like**
A method "restart" or "search" to specific tick

**Describe alternatives you've considered**
Is it possible to "restart" the buffer so it would have the same effect?

**Comment by saul** [comment, marked as answer]
You can restart the demo by calling `DemoFile#parse` again with the same buffer. Unfortunately you can't 'skip' to a tick as every tick builds on the previous tick. It would also be quite difficult to be able to rewind as that would require you remembering the state of the world in previous ticks (wasting memory).

This is the reason why `demo_gototick` in CSGO is relatively slow. It can't just jump to the tick, it has to process every tick before then as quickly as it can. Similarly this is why you can't move backwards in CSGO demos without it having to jump to the start of the demo and 'fast forwarding'.

As for how you could emulate `demo_gototick` with this library, I suggest listening to the `tickstart` event and waiting for the tick you're interested in to start parsing. At this point you can add your event listeners. This won't be super fast but unfortunately there's no way to speed it up. The library needs to process every tick to build up its model of the game world.

**Comment by Henroks** [comment]
Calling the DemoFile#parse on the same buffer "breaks" the replay (Events are not getting triggered anymore)

**Comment by saul** [comment]
Did you `cancel()` first?

**Comment by Henroks** [comment]
Yes
